test/runner/e2e_read_test.cpp

```diff
@@ -3,159 +3,69 @@
-};
-
-TEST_F(LongStringPKTest, LongStringPKTest) {
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/long_string_pk/long_string_pk.test"));
-}
-
-TEST_F(TinySnbReadTest, Match) {
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/match/node.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/match/one_hop.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/match/two_hop.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/match/three_hop.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/match/four_hop.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/match/multi_query_part.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/match/multi_label.test"));
-}
-
-TEST_F(TinySnbReadTest, Filter) {
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/filter/node.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/filter/one_hop.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/filter/two_hop.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/filter/four_hop.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/filter/five_hop.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/filter/multi_query_part.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/filter/multi_label.test"));
-}
+    std::string getInputDir() override { return TestHelper::appendKuzuRootPath(dataset_); }
 
-TEST_F(TinySnbReadTest, AccHJ) {
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/acc/acc_hj.test"));
-}
-
-TEST_F(TinySnbReadTest, Function) {
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/function/offset.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/function/date.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/function/timestamp.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/function/interval.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/function/list.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/function/arithmetic.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/function/boolean.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/function/string.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/function/cast.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/function/case.test"));
-}
+    explicit RunDynamicTest(std::vector<std::string> testFiles, std::string dataset)
+        : testFiles_(testFiles), dataset_(dataset) {}
```

Comment:
wdym by `move` both `testFiles` and `dataset`?

Reply:
I mean you should use `std::move` here to avoid copy:
```
: testFiles(std::move(testFiles)), dataset(std::move(dataset)) {}
```

You don't need to add the "\_" postfix to class fields. We usually don't do that.
When there is shadowing of class variable names in function params, we usually add "_" postfix to function params.